records_mover/records/delimited/types.py

```diff
@@ -71,3 +58,44 @@
                    "timeonlyformat",
                    "datetimeformattz",
                    "datetimeformat"]
+
+HINT_NAMES: List[HintName] = list(get_args(HintName))  # type: ignore
+
+# TODO: Retire this
+BootstrappingRecordsHints = TypedDict('BootstrappingRecordsHints',
+                                      {
+                                          'quoting': HintQuoting,
+                                          'header-row': HintHeaderRow,
+                                          'field-delimiter': HintFieldDelimiter,
+                                          'encoding': HintEncoding,
+                                          'escape': HintEscape,
+                                          'compression': HintCompression,
+                                          'record-terminator': HintRecordTerminator,
+                                      },
+                                      total=False)
+
+# TODO: Retire this
+BOOTSTRAPPING_HINT_NAMES: List[HintName] = typed_dict_keys(BootstrappingRecordsHints)
+
+PartialRecordsHints = TypedDict('PartialRecordsHints',
+                                {
+                                    'quoting': HintQuoting,
+                                    'header-row': HintHeaderRow,
+                                    'field-delimiter': HintFieldDelimiter,
+                                    'encoding': HintEncoding,
+                                    'escape': HintEscape,
+                                    'compression': HintCompression,
+                                    'record-terminator': HintRecordTerminator,
+                                    'quotechar': HintQuoteChar,
+                                    'doublequote': HintDoublequote,
+                                    'dateformat': HintDateFormat,
+                                    'timeonlyformat': HintTimeOnlyFormat,
+                                    'datetimeformattz': HintDateTimeFormatTz,
+                                    'datetimeformat': HintDateTimeFormat,
+                                },
+                                total=False)
+# TODO: Rename this throughout?
```

Comment:
Uncompleted punchlist item detected--consider resolving or moving this to your issue tracker